main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+)
+
+const base36 = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+func addZeros(a string, time int) string {
+	for i := time; i > 0; i-- {
+		a = "0" + a
+	}
+	return a
+}
+
+func Normalize(a, b string) (string, string) {
+	if len(a) > len(b) {
+		b = addZeros(b, len(a)-len(b))
+	} else {
+		a = addZeros(a, len(b)-len(a))
+	}
+	return a, b
+}
+
+func LongAdd(a, b string) string {
+	var result string
+	var add uint8
+	a, b = Normalize(a, b)
+	for i := len(a) - 1; i >= 0; i-- {
+		num := (a[i] - '0') + (b[i] - '0') + add
+		add = num / 10
+		result = string((num%10)+'0') + result
+	}
+	if add != 0 {
+		result = "1" + result
+	}
+	return result
+}
+
+func Byte2Number(a byte) int {
+	return strings.Index(base36, string(a))
+}
+
+func LongAddWithBase(a, b string, base int) string {
+	var result string
+	var add int
+	a, b = Normalize(a, b)
+	for i := len(a) - 1; i >= 0; i-- {
+		num := Byte2Number(a[i]) + Byte2Number(b[i]) + add
+		add = num / base
+		result = string(base36[num%base]) + result
+	}
+	if add != 0 {
+		result = "1" + result
+	}
+	return result
+}
+func LongFibonacci(n int) string {
+	if n == 0 {
```

Comment:
nit: this code could be slightly shortened by removing this extra check:
```go
a, b := "0", "1"
for ; n > 0; n-- {
  prev, now = ...
}
return prev
```
as you can see, it does the same, but becomes almost twice shorter